### DISCUSSION TITLE
Conditional fields not working properbly

### DISCUSSION THREAD

**Discussion by ghost** [react-hook-form/react-hook-form#2107] [Q&A] [answered]
**Describe the bug**
Conditional field gives wrong values.

**To Reproduce**

*Example 1*
1. select status 1
expected -> {status: "1", details: ""}
received -> {status: "1"}
2. select status 2
expected -> {status: "2"}
received -> {status: "2", details: ""}

*Example 2*
1. select status 1
2. write into details "123"
expected -> {status: "1", details: "123"}
received -> {status: "1", details: "123"}
3. select status 2
expected -> {status: "2"}
received -> {status: "2", details: "123"}

**Live Example**
https://stackblitz.com/edit/react-ts-rtdcud

**Code**

```typescript
import React, { Component } from 'react';
import { render } from 'react-dom';
import {useForm} from'react-hook-form'

const App = () => {
  const {handleSubmit, register, getValues} = useForm();

  return (
    <form onChange={handleSubmit(data => console.log(data))}>

      <label>Status</label>

      <br/>

      <select ref={register} name='status'>
        <option>-</option>
        <option>1</option>
        <option>2</option>
      </select>

      <br/><br/>

      <label>Details</label>
      
      <br/>

      {getValues().status === '1' && <input ref={register} name='details' />}
    </form>
  );

}

render(<App />, document.getElementById('root'));
```

**Desktop**
 - OS: Windows 10
 - Browser Chrome
 - Version 83.0.4103.116

**Comment by pmaier983** [comment, marked as answer]
In this case I think it might help to use [`useWatch`](https://react-hook-form.com/api#useWatch) instead of getValues, and to analyze the formSubmit only on submit! Also take a look at [`unregister`](https://react-hook-form.com/api#unregister) for removing a property from your submit 😄

**Reply by ghost** [reply]
Hi pmaier983,

`useWatch` / `watch` instead of `getValues` was what I needed because it triggers a re-render. I updated the stackblitz. Now everything is working as expected! Thank you! 😄

**New Code**
```typescript
import React, { Component, useEffect } from 'react';
import { render } from 'react-dom';
import {useForm, useWatch} from'react-hook-form'

const App = () => {
  const {handleSubmit, register, control} = useForm();
  const watch = useWatch({control});

  return (
    <form onSubmit={handleSubmit(data => console.log(data))}>

      <label>Status</label>

      <br/>

      <select ref={register} name='status'>
        <option>-</option>
        <option>1</option>
        <option>2</option>
      </select>

      <br/><br/>

      <label>Details</label>
      
      <br/>

      {watch.status === '1' && <input ref={register} name='details' />}

      <br/><br/>

      <input type='submit'/>
    </form>
  );

}

render(<App />, document.getElementById('root'));
```
